### PR TITLE
fix the consumer remoteApplication is null

### PR DESCRIPTION
**Pull request by xixingya**
## What is the purpose of the change

https://github.com/apache/dubbo/issues/12631

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [x ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).
fix the consumer remoteApplicationName is null

### CODE REVIEW

**Comment by xixingya** [comment]
fix https://github.com/apache/dubbo/issues/12631

File: dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/RpcServiceContext.java [hunk 1]
```diff
@@ -201,6 +202,10 @@ public URL getUrl() {
     @Override
     public void setUrl(URL url) {
         this.url = url;
+        if (url.getSide().equals(CommonConstants.CONSUMER)) {
+            this.remoteApplicationName = url.getRemoteApplication();
+            this.setRemoteAddress(url.getHost(), url.getPort());
+        }
```
Comment (AlbumenJ) [inline dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/RpcServiceContext.java:208]: image

Set remote attribute in `org.apache.dubbo.rpc.cluster.support.AbstractClusterInvoker#invokeWithContext` would be a better choice.
```
    private static void setRemote(Invoker<?> invoker, Invocation invocation) {
        invocation.addInvokedInvoker(invoker);
        RpcContext.getServiceContext().setRemoteAddress(invoker.getUrl().toInetSocketAddress());
        RpcContext.getServiceContext().setRemoteApplicationName(invoker.getUrl().getRemoteApplication());
    }
```

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/DubboInvoker.java [hunk 2]
```diff
@@ -107,8 +107,6 @@ protected Result doInvoke(final Invocation invocation) throws Throwable {
 
             invocation.setAttachment(TIMEOUT_KEY, String.valueOf(timeout));
 
-            RpcContext.getServiceContext().setRemoteAddress(currentClient.getRemoteAddress());
```
Comment (AlbumenJ) [inline dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/DubboInvoker.java:110]: image

Check style should be fixed.

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/DubboInvoker.java (hunk 2 above)
Comment (xixingya) [reply dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/DubboInvoker.java:110]: thanks,I will fix

File: dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/RpcServiceContext.java (hunk 1 above)
Comment (xixingya) [reply dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/RpcServiceContext.java]: @AlbumenJ hello, I have fix them,and I run the local test on filter  can get them

**Review by AlbumenJ — APPROVED** [review]

**Comment by AlbumenJ** [comment]
LGTM. Thx your contributions~

**Comment by sadeychai** [comment]
@AlbumenJ 3.1.8也遇到此问题，3.1.x是否有修复计划？

**Pull request merged**